File: 0004-寻找两个有序数组的中位数/Solution.java
class Solution {
    public double findMedianSortedArrays(int[] nums1, int[] nums2) {
        int[] helper = new int[nums1.length + nums2.length];
        int index = 0, i, j;
        for(i=0, j=0; i<nums1.length && j<nums2.length ;){
            if(nums1[i] < nums2[j]){
                helper[index++] = nums1[i++];
            }else{
                helper[index++] = nums2[j++];
            }
        }
        if(i == nums1.length){
            while(j < nums2.length){
                helper[index++] = nums2[j++];
            }
        }else{
            while(i < nums1.length){
                helper[index++] = nums1[i++];
            }
        }

        // 这里要除以2.0, 确保得到的结果为double类型
        if( helper.length % 2 == 0){
            return (helper[helper.length/2] + helper[helper.length/2 - 1]) / 2.0;
        }
        return helper[helper.length/2];
    }


    public static void main(String[] args){
        int[] nums1 = new int[]{1, 2};
        int[] nums2 = new int[]{3, 4};
        System.out.println(new Solution().findMedianSortedArrays(nums1, nums2));
    }
}

File: 0004-寻找两个有序数组的中位数/Solution.py
class Solution:
    def findMedianSortedArrays(self, nums1, nums2):
        # extend 比 nums1+nums2 快很多
        nums1.extend(nums2)
        helper = nums1
        helper.sort()
        if len(helper)%2 == 0:
            # //为取整
            return (helper[len(helper)//2] + helper[len(helper)//2-1]) / 2
        return helper[len(helper)//2]





if __name__ == "__main__":
    print(Solution().findMedianSortedArrays([1,3], [2]))    